fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalUnion.java

```diff
@@ -210,7 +217,54 @@ public void computeUnique(DataTrait.Builder builder) {
 
     @Override
     public void computeUniform(DataTrait.Builder builder) {
-        // don't propagate uniform slots
+        CascadesContext tempCascadeContext = CascadesContext.initContext(
+                ConnectContext.get().getStatementContext(), this, PhysicalProperties.ANY);
+        ExpressionRewriteContext context = new ExpressionRewriteContext(tempCascadeContext);
```

Comment:
if just need to do constant folding, could we not relay on CascadesContext? i saw only few function relay on `CascadesContext`, and them should be folding in the very beginning, so maybe we could pass a `null` to `ExpressionRewriteContext` and let FoldConstantByFE be safe for a null pointer `CascadesContext`

Reply:
and furthermore, could `checkConstantExpr` do not relay on constant folding? if it need a literal, we should do it apply enough expression rewrite before use it